cocotb/share/lib/fli/FliCbHdl.cpp

```diff
@@ -140,6 +144,57 @@ int FliStartupCbHdl::arm_callback()
     return 0;
 }
 
+
+static void get_args(gpi_sim_info_t *info)
+{
+    /*  Necessary to implement PLUSARGS
+        There is no function available on the FLI to obtain argc+argv directly from the
+        simulator. To work around this we use the TCL interpreter that ships with Questa,
+        some TCL commands, and the TCL variable `argv` to obtain the simulator argc/argv.
+    */
+    info->argc = 0;
+    info->argv = NULL;
+
+    Tcl_Interp* interp = reinterpret_cast<Tcl_Interp*>(mti_Interp());
+
+    // get argc
+    if (mti_Cmd("llength $argv") != TCL_OK) {
+        return;
+    }
+    const char* result = Tcl_GetStringResult(interp);
+    unsigned long argc_u = strtoul(result, NULL, 10);
+    Tcl_ResetResult(interp);
+    if (argc_u > std::numeric_limits<int>::max()) {
+        return;
+    }
+    int argc = static_cast<int>(argc_u);
+    if (argc_u == 0) {
+        return;
+    }
+
+    // allocate storage for argv
+    char** argv = new char*[argc];
```

Comment:
To avoid the appearance leaking, can you declare this a `static vector<std::string>`, and unpack the pointers into the return value?

The value shouldn't change at runtime anyway, so `static` seems appropriate.

Reply:
It doesn't even need to be `static`, it's no longer used after the call to `gpi_embed_init`, I just wasn't cleaning up afterwards.

Reply:
Right, but the design of the GPI doesn't even allow you to clean up, does it? Other implementations return non-owned pointers.

Reply:
Well it travels up from`gpi_embed_init` to `embed_sim_init`, it gets copied into a Python list and  returns back down the stack. No one holds a reference to it elsewhere. So you could clean up after the call to `gpi_embed_init` in all of the interfaces. You'll see that VHPI cleans up.